code/game/objects/items/shuttle_creator.dm

```diff
@@ -0,0 +1,253 @@
+#define SHUTTLE_CREATOR_MAX_SIZE 200
+#define CUSTOM_SHUTTLE_LIMIT 4
+
+/obj/item/shuttle_creator
+	name = "Rapid Shuttle Designator"
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "rsd"
+	lefthand_file = 'icons/mob/inhands/equipment/tools_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/tools_righthand.dmi'
+	desc = "A device used to define the area required for custom ships. Uses bluespace crystals to create bluespace-capable ships."
+	density = FALSE
+	anchored = FALSE
+	flags_1 = CONDUCT_1
+	item_flags = NOBLUDGEON
+	force = 0
+	throwforce = 8
+	throw_speed = 3
+	throw_range = 5
+	w_class = WEIGHT_CLASS_NORMAL
+	req_access_txt = "11"
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 50)
+	resistance_flags = FIRE_PROOF
+	var/static/customShuttles = 0
+	var/ready = TRUE
+	var/recorded_shuttle_area
+	var/list/loggedTurfs = list()
+	var/loggedOldArea
+	var/linkedShuttleId
+
+/obj/item/shuttle_creator/attack_self(mob/user)
+	..()
+	if(linkedShuttleId)
+		return
+	if(customShuttles > 4)
+		return
+	return check_current_area(user)
+
+/obj/item/shuttle_creator/afterattack(atom/target, mob/user, proximity_flag)
+	. = ..()
+	if(!ready)
+		to_chat(user, "<span class='warning'>You need to define a shuttle area first.</span>")
+		return
+	if(!proximity_flag)
+		return
+	if(istype(target, /obj/machinery/door/airlock))
+		if(linkedShuttleId)
+			return
+		if(customShuttles > 4)
+			to_chat(user, "<span class='warning'>Shuttle limit reached, sorry.</span>")
+			return
+		if(!create_shuttle_area(user))
+			return
+		if(shuttle_create_docking_port(target, user))
+			to_chat(user, "<span class='notice'>Shuttle created!</span>")
+		return
+	else if(istype(target, /obj/machinery/computer/custom_shuttle))
+		if(!linkedShuttleId)
+			to_chat(user, "<span class='warning'>Error, no defined shuttle linked to device</span>")
+			return
+		var/obj/machinery/computer/custom_shuttle/console = target
+		console.linkShuttle(linkedShuttleId)
+		to_chat(user, "<span class='notice'>Console linked successfully!</span>")
+		return
+	else if(istype(target, /obj/machinery/computer/camera_advanced/shuttle_docker/custom))
+		if(!linkedShuttleId)
+			to_chat(user, "<span class='warning'>Error, no defined shuttle linked to device</span>")
+			return
+		var/obj/machinery/computer/camera_advanced/shuttle_docker/custom/console = target
+		console.linkShuttle(linkedShuttleId)
+		to_chat(user, "<span class='notice'>Console linked successfully!</span>")
+		return
+	to_chat(user, "<span class='warning'>The [src] bleeps. Select an airlock to create a docking port, or a valid machine to link.</span>")
+	return
+
+/obj/item/shuttle_creator/proc/calculate_bounds(obj/docking_port/mobile/port)
```

Comment:
This method probably doesn't work at the moment and makes the bounds from a weird point
The size is done, but it needs docking offsets to be set correctly for it to work properly.

Reply:
You can likely figure out the docking offsets by getting the type of each turf going outward from the docking port and iterating an integer until it becomes space or a different area, in both the height direction and width direction.

Reply:
Thanks, you are awesome